enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+as a standard network adapter within user space. This means that no additional configuration is required within RHCOS
+to allow EFA instances to be used as regular OpenShift nodes, existing networking is unaffected.
+
+To take advantage of the EFA performance benefits, additional libfabric software must be deployed.
+This software is currently [distributed by AWS](https://github.com/aws-samples/aws-efa-eks/blob/main/manifest/efa-k8s-device-plugin.yml)
+and it will be up to the end user to deploy this software within their clusters.
+Red Hat will not initially support the libfabric driver.
+
+#### Only a subset of instance types are supported with EFA network interfaces
+
+Only a small subset of AWS instance types (eg. m5dn.24xlarge, m5dn.metal) are supported for use with EFA interfaces.
+If a user were to specify an EFA interface type with a non supported instance (eg. m5.xlarge), then the instance will
+fail to create. In this scenario, AWS rejects the RunInstance call from Machine API which causes Machine API to mark the
+Machine as `Failed`. An error message will be added to the Machine status to signal to the user that their configuration
+was invalid.
+
+> error launching instance: EFA interfaces are not supported on m5.2xlarge
+
+## Design Details
+
+### Open Questions
+
+- What support will we provide in the future?
+  - Will we be able to provide full end to end support for the required libfabric drivers?
+- How can the feature be used by end users?
+  - How do they signal to OpenShift that a particular workload should leverage the EFA feature?
+
+### Test Plan
+
+In the first phase, we are planning to just provide users the ability to create Machines with the EFA interface type.
+We will run standard regression testing on the feature but we do not expect any special additional testing to be
+performed.
+
+In the future, if/when we introduce full support for this feature, we will need to introduce specific testing to check
+that the workloads can leverage the EFA feature and that their networking throughput is increased when leveraging the
+feature.
+
+### Graduation Criteria
+
+The addition of an API field to Machine API implies that the feature is GA from the beginning, no graduation criteria
+are required.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+No features will be removed as a part of this proposal
+
+### Upgrade / Downgrade Strategy
+
+Existing clusters being upgraded will not be configured for the new feature and will continue to use the existing
+standard interface type.
+
+Once configured, on downgrade, the Machine API components will not know about the new fields, and as such, will ignore
+the interface type field if specified. The usage of an EFA interface type will not affect removal of Machines after a
+downgrade, there should be no detrimental effect of a downgrade on Machine API.
```

Comment:
Will the machines be removed or would you expect that the machines already created would "coast"?

Reply:
what downgrade? by whom?